File: Backend/models/Prerequisite.js

// models/Prerequisite.js
const Sequelize = require('sequelize');
const sequelize = require('../config/database');
const Course = require('./Course');
const Prerequisite = sequelize.define('CoursePrerequisite', {
  course_id: {
    type: Sequelize.STRING(10),
    primaryKey: true
  },
  prerequisite_course_id: {
    type: Sequelize.STRING(10),
    primaryKey: true
  }
}, {
  timestamps: false, // Disable automatic timestamps
  tableName: 'CoursePrerequisites'
});

module.exports = Prerequisite;

File: grade-simulator-service/dummy_data/models/prerequisite.py
from sqlalchemy import Column, String, Integer, ForeignKey, Table
from db import Base

class Prerequisite(Base):
    __tablename__ = "CoursePrerequisites"
    
    course_id = Column(String(10), ForeignKey("Courses.course_id"), primary_key=True, nullable=False)
    prerequisite_course_id = Column(String(10), ForeignKey("Courses.course_id"), primary_key=True, nullable=False)
    
    def to_dict(self):
        return {
            "course_id": self.course_id,
            "prerequisite_course_id": self.prerequisite_course_id
        } 